### ISSUE TITLE
php7.2-igbinary  broken on Debian 9

### ISSUE THREAD

**Issue by toubsen** [oerdnj/deb.sury.org#1626]
**Frequently asked questions**

- [x] I have read [Frequenty Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Describe the bug**
Serializing + unserializing a nested array with igbinary will result in a different pointer structure in the unserialized array, that breaks e.g. the `current()` function.

Broken package version: `php7.2-igbinary: 3.2.4+2.0.8-1+0~20210802.26+debian9~1.gbpfc46c4`
Non-broken package version: `php7.2-igbinary: 3.2.1+2.0.8-6+0~20210223.25+debian9~1.gbp4f4993`

**To Reproduce**
1. Run minimal reproducer PHP script (e.g. from CLI)

```
$arr = [
	'one' => [
		'two' => [
			'three' => [
				'four'
			],
		],
	],
];
$test = current(current(current(current($arr))));

echo $test . "\n";

$arr2 = igbinary_unserialize(igbinary_serialize($arr));
$test2 = current(current(current(current($arr2))));

echo $test2 . "\n";
```

**Expected behavior**
Original and serialized + unserialized array should behave the same

Expected output from reproducer:
```
four
four
```

Actual output from reproducer:
```
four
PHP Warning:  current() expects parameter 1 to be array, boolean given in foo.php on line 16

Warning: current() expects parameter 1 to be array, boolean given in foo.php on line 16
PHP Warning:  current() expects parameter 1 to be array, null given in foo.php on line 16

Warning: current() expects parameter 1 to be array, null given in foo.php on line 16
PHP Warning:  current() expects parameter 1 to be array, null given in foo.php on line 16

Warning: current() expects parameter 1 to be array, null given in foo.php on line 16
```

**Distribution (please complete the following information):**
 - OS: Debian GNU/Linux 9.13 (stretch)
 - Architecture: amd64
 - Repository: packages.sury.org

**Package(s) (please complete the following information):**
```
php7.2-igbinary:
  Installed: 3.2.4+2.0.8-1+0~20210802.26+debian9~1.gbpfc46c4
```

**Additional context**
The PHP 8 version of the package with the same build identifier is not broken:
`php8.0-igbinary: 3.2.4+2.0.8-1+0~20210802.26+debian9~1.gbpfc46c4`

**Comment by oerdnj** [comment]
This is not a bug in the package, but rather in the upstream version. You need to report the issue to the igbinary developers.

**Comment by toubsen** [comment]
Looked up Readme.Debian.gz for all versions, and affected version (PHP 7.2) and unaffected version (PHP 8.0) are both build on the same git commit.

**Comment by stronk7** [comment]
Just noting that this has been fixed upstream:

> 3.2.5: Fix change in behavior introduced in 3.2.2RC1 when unserializing arrays - the internal array pointer (for `next()`, `key()`, etc) pointed past the end of the array in php 7.0-7.2.

([source](https://github.com/igbinary/igbinary/releases/tag/3.2.5))

It would be great to get a new version of the extension for the affected versions rolled.
